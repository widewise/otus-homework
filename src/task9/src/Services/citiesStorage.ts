import {ICityWeather} from "../Models/weather";
import {Error} from "../components/Error";

const CITIES_KEY = "CITIES";

function getCities(): ICityWeather[] {
    const jsonData = localStorage.getItem(CITIES_KEY);
    if (!jsonData) {
        throw Error("Storage is not initialized");
    }
    return  JSON.parse(jsonData) as ICityWeather[];
}

export const citiesStorage = {
    init: (cities: ICityWeather[]) => {
        localStorage.setItem(CITIES_KEY, JSON.stringify(cities));
    },

    searchCities: (search: string = ""): ICityWeather[] => {
        return getCities().filter(c => c.name.includes(search));
    },

    getCity: (id: string): ICityWeather | undefined => {
        const city = getCities().find(c => c.id === id);
        if(!city) {
            console.log(`City with id ${id} doesn't exist`);
            return;
        }
        return city;
    }
}