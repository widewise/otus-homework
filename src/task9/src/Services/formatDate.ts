const padTo2Digits = (num: number) => {
    return num.toString().padStart(2, '0');
}

export function formatDate(date: Date): string {
    const aDate = new Date(date);
    if(!aDate) {
        return "";
    }

    return `${padTo2Digits(aDate.getDate())}:${padTo2Digits(aDate.getMonth() + 1)}:${aDate.getFullYear()}`;
}