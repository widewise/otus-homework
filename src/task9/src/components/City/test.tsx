import React from "react";
import renderer from "react-test-renderer";
import { City } from "./index";
import { MemoryRouter } from "react-router";
import { useParams } from "react-router";
import { citiesStorage } from "../../Services/citiesStorage";

jest.mock("react-router");
jest.mock("../../Services/citiesStorage");

describe(`${City.name}`, () => {
    const city = {
        id: "1",
        name: "city",
        data: []
    };
    let actual: any;

    describe(`rerender with data ${City.name}`, () => {
        beforeEach(() => {
            //arrange
            (useParams as jest.Mock).mockImplementationOnce(() => { "1" });
            (citiesStorage.getCity as jest.Mock).mockImplementationOnce(() => city);

            //act
            actual = renderer.create(<MemoryRouter><City /></MemoryRouter>).toJSON();
        });

        it('should match snapshot', () => {
            //assert
            expect(actual).toMatchSnapshot();
        });
    });

    describe(`rerender without data ${City.name}`, () => {
        beforeEach(() => {
            //act
            actual = renderer.create(<MemoryRouter><City /></MemoryRouter>).toJSON();
        });

        it('should match snapshot', () => {
            //assert
            expect(actual).toMatchSnapshot();
        });
    });
});