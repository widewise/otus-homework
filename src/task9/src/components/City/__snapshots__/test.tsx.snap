// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City rerender with data City should match snapshot 1`] = `null`;

exports[`City rerender without data City should match snapshot 1`] = `null`;
